src/main/java/com/github/manolo8/darkbot/core/objects/facades/QuestProxy.java

```diff
@@ -0,0 +1,300 @@
+package com.github.manolo8.darkbot.core.objects.facades;
+
+import com.github.manolo8.darkbot.core.itf.Updatable;
+import com.github.manolo8.darkbot.core.objects.swf.FlashList;
+import com.github.manolo8.darkbot.core.objects.swf.ObjArray;
+import com.github.manolo8.darkbot.core.utils.ByteUtils;
+
+import eu.darkbot.api.API;
+
+import static com.github.manolo8.darkbot.Main.API;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jetbrains.annotations.Nullable;
+
+public class QuestProxy extends Updatable implements API.Singleton {
+    private @Nullable Quest currentQuest;
+
+    private final List<QuestListItem> questItems = new ArrayList<>();
+
+    private final ObjArray questItemsArr = ObjArray.ofArrObj(true);
+
+    private @Nullable Quest questGiverSelected;
+    private @Nullable QuestListItem questInfoGiverSelected;
+
+    private boolean visibleQuestGiver;
+    private int tabSelected;
+
+    @Override
+    public void update() {
+        if (address == 0) {
+            return;
+        }
+
+        long questClass = API.readMemoryLong(address + 0x98) & ByteUtils.ATOM_MASK;
```

Comment:
```suggestion
        long questClass = API.readMemoryPtr(address + 0x98);
```
try to use readMemoryPtr to read pointers instead of adding the & mask at the end

Reply:
Fixed